src/Common/src/System/Net/NetworkInformation/UnixCommandLinePing.cs

```diff
@@ -37,6 +41,27 @@ private static string GetPingUtilityPath(bool ipv4)
             return null;
         }
 
+        // Check if found ping is symlink to busybox like alpine /bin/ping -> /bin/busybox
+        private static unsafe bool IsBusyboxPing(string pingBinary)
+        {
+            byte* buffer = stackalloc byte[4000];
+            int resultLength = ReadLink(pingBinary, buffer, 4000);
+
+            // If pingBinary is not link resultLength will be -1
+            if (resultLength > 0 && resultLength < 4000)
```

Comment:
Where is this 4000 number coming from?  It's used multiple times, so it'd be good to put it in a named constant, but I'm unclear as to its origin.  Is it guaranteed to be big enough?  Is it larger than is actually needed?  Do we need to try again with ReadLink if the buffer size was insufficient?

Reply:
My memory failed me. I was thinking about MAXPATHLEN but that is only 1024 on Linux. That should be sufficient for any case. And probably bigger than we need as well. It is always possible that some system or particular installation would do something weird with ping and we may fail. When ping is part of os and it is link to something else, it would normally live in /[usr/]/[local]/[s]bin If it is not, it is probably not busybox. 
Also note, that the code does not (and cannot) detect hard links. It will detect normal scenarios like Alpine and it may fail to detect corner cases.